src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -30,6 +30,68 @@ public void Process(Exception exception, SentryEvent sentryEvent)
         sentryEvent.SentryExceptions = sentryExceptions;
     }
 
+    // Sentry exceptions are sorted oldest to newest.
+    // See https://develop.sentry.dev/sdk/event-payloads/exception
+    internal IReadOnlyList<SentryException> CreateSentryExceptions(Exception exception)
+    {
+        var exceptions = WalkExceptions(exception).Reverse().ToList();
+
+        // In the case of only one exception, ExceptionId and ParentId are useless.
+        if (exceptions.Count == 1 && exceptions[0].Mechanism is { } mechanism)
+        {
+            mechanism.ExceptionId = null;
+            mechanism.ParentId = null;
+            if (mechanism.IsDefaultOrEmpty())
+            {
+                // No need to convey an empty mechanism.
+                exceptions[0].Mechanism = null;
+            }
+        }
+
+        return exceptions;
+    }
+
+    private class Counter
+    {
+        private int _value;
+
+        public int GetNextValue() => _value++;
```

Comment:
I got confused by this. The counter starts at zero. So when the first time `GetNextValue()` is invoked, should it be `1`? But it will return 0. I'm assuming this is by design but is worth asking 😊

Reply:
By design. It's a postfix, so the value returned is before the increment.

https://learn.microsoft.com/dotnet/csharp/language-reference/operators/arithmetic-operators#increment-operator-

Ultimately this class is just an easy way to assign an ID to each item, starting with 0.